test/sketch_test.cpp

```diff
@@ -5,7 +5,7 @@
 
 TEST(SketchTestSuite, TestExceptions) {
   Sketch sketch = Sketch(10,rand());
-  ASSERT_THROW(sketch.query(), NoGoodBucketException);
```

Comment:
can you write another test to check `NoGoodBucketException`?

Reply:
Alright I wrote it, anything else we need?

Reply:
looks good, lgtm